modules/server/src/network/aggregations/AggregationAccumulator.ts

```diff
@@ -37,13 +47,19 @@ const resolveAggregations = ({ data, accumulator, requestedFields }: ResolveAggr
 		const { aggregations, hits } = data;
 
 		const isFieldAvailable = !!aggregations[requestedField];
-		const type = ALL_NETWORK_AGGREGATION_TYPES_MAP.get(requestedField);
+
+		/*
+		 * requested field will always be in ALL_NETWORK_AGGREGATION_TYPES_MAP
+		 * GQL schema validation will throw an error earlier if a requested field isn't in the schema
+		 */
+		const type = ALL_NETWORK_AGGREGATION_TYPES_MAP.get(requestedField) as SupportedAggregation;
 		const existingAggregation = accumulator[requestedField];
 
 		if (isFieldAvailable) {
 			accumulator[requestedField] = addToAccumulator({
 				existingAggregation,
-				aggregation: aggregations[requestedField],
+				// similarly due to GQL schema validation this will not be undefined
+				aggregation: aggregations[requestedField]!,
```

Comment:
To remove this non-null assertion we can store `aggregations[requestedField]` in a variable and test if that value is defined, vs using the `isFieldAvailable` bool:

```ts
const aggregation = aggregations[requestedField];
		if (aggregation !== undefined) {
			accumulator[requestedField] = addToAccumulator({
				existingAggregation,
				// similarly due to GQL schema validation this will not be undefined
				aggregation,
				type,
			});
		} else {
```